extensions/sdk/src/Rally.ts

```diff
@@ -137,6 +129,21 @@ export class Rally {
     ["source", "medium", "campaign", "term", "content"].forEach((key) => {
       attribution[key] = url.searchParams.get(`utm_${key}`);
     });
+
+    return attribution;
+  }
+
+  /**
+   * Attempt to fetch the attribution codes from the store page URL for this extension.
+   */
+  private async storeAttributionCodes() {
+    let attribution = await this.getAttributionCodes();
+    if (!(Object.keys(attribution).length === 0 && attribution.constructor === Object)) {
```

Comment:
As I understand getAttributionCodes would always return an object right? If so then we can remove the constructor check and invert the condition on length to make it more readable:

```
if (Object.keys(attribution).length) {
   // return;
}
```

Another question: are we sure we want to do early bailout here? Is there a situation where codes are present and values aren't (or need to be changed?)

Reply:
I'll try that thanks!

> Another question: are we sure we want to do early bailout here? Is there a situation where codes are present and values aren't (or need to be changed?)

I thought about this a bit, but I can't think of any situation where we would want to change the attribution codes, they are supposed to represent how the user got to the extension store. If the user were to go back to the extension store page following some other link I wouldn't want the attribution codes to be reset.